src/csharp/Microsoft.Spark.Worker.UnitTest/SimpleWorkerTests.cs

```diff
@@ -25,7 +26,16 @@ public void TestsSimpleWorkerTaskRunners(string version)
             Task clientTask = Task.Run(() => simpleWorker.Run(ipEndpoint.Port));
 
             PayloadWriter payloadWriter = new PayloadWriterFactory().Create(typedVersion);
-            TaskRunnerTests.TestTaskRunnerReadWrite(serverListener, payloadWriter);
+            ISocketWrapper serverSocket = serverListener.Accept();
+
+            if (payloadWriter.Version.Major >= 3 && payloadWriter.Version.Minor >= 2)
```

Comment:
This doesn't cover the case of something like Spark 4.0.0.

Reply:
Changed